src/main/java/com/algorand/algosdk/transaction/Transaction.java

```diff
@@ -992,6 +993,11 @@ public AssetParams(
                 if (metadataHash.length > 32) throw new RuntimeException("asset metadataHash cannot be greater than 32 bytes");
                 this.metadataHash = metadataHash;
             }
+
+            if(metadataHash != null) {
+                if (!Base64.isBase64(metadataHash)) throw new RuntimeException("asset metadataHash is not base64 encoded");
```

Comment:
Please also add the metadataHash to the exception message. It will be easier to identify with metadataHash is not well-formed.

Reply:
please move this into the previous `metadataHash != null` check

Reply:
Added metadataHash to the error message and combined blocks together.